src/components/Table/Table.tsx

```diff
@@ -0,0 +1,1100 @@
+	cellType: Th | Td = Td,
+	mapFn: (gridcell: IGridCell, ...args: any[]) => any = _.property('element')
+) {
+	const cellRowList = _.map(trList, trElement =>
+		_.map(filterTypes(trElement.props.children, cellType))
+	);
+	const grid: IGridCell[][] = [];
+
+	if (_.isEmpty(cellRowList)) {
+		return [];
+	}
+
+	// iterate over each row
+	for (let rowIndex = 0; rowIndex < cellRowList.length; rowIndex++) {
+		const cellRow = cellRowList[rowIndex];
+
+		if (_.isNil(grid[rowIndex])) {
+			grid[rowIndex] = [];
+		}
+
+		const canonicalRow = rowIndex;
+
+		// build out each horizonal duplicates of each cell
+		for (
+			let cellElementIndex = 0;
+			cellElementIndex < cellRow.length;
+			cellElementIndex++
+		) {
+			const cellElement = cellRow[cellElementIndex];
+
+			let colSpan = 1;
+			let isCellIncluded = false;
+
+			if (_.isNumber(cellElement.props.colSpan)) {
+				colSpan = cellElement.props.colSpan;
+			}
+
+			const nilCellIndex = _.findIndex(grid[canonicalRow], _.isNil);
+			const originCol =
+				nilCellIndex !== -1 ? nilCellIndex : grid[canonicalRow].length;
+
+			for (let currentColSpan = 0; currentColSpan < colSpan; currentColSpan++) {
+				grid[canonicalRow][originCol + currentColSpan] = {
+					element: cellElement,
+					canonicalPosition: {
+						row: canonicalRow,
+						col: originCol,
+					},
+					isOriginal: !isCellIncluded,
+				};
+				isCellIncluded = true;
+			}
+		}
+
+		// build out each vertial duplicates of each cell using the new row in the full grid
+		for (let colIndex = 0; colIndex < grid[canonicalRow].length; colIndex++) {
+			const gridCell = grid[canonicalRow][colIndex];
+			if (gridCell.isOriginal) {
+				const cellElement = _.get(gridCell, 'element');
+				let rowSpan = 1;
+
+				if (_.isNumber(_.get(cellElement, 'props.rowSpan'))) {
+					rowSpan = _.get(cellElement, 'props.rowSpan');
+				}
+
+				for (
+					let currentRowSpan = 1;
+					currentRowSpan < rowSpan;
+					currentRowSpan++
+				) {
+					if (_.isNil(grid[canonicalRow + currentRowSpan])) {
+						grid[canonicalRow + currentRowSpan] = [];
+					}
+
+					grid[canonicalRow + currentRowSpan][colIndex] = _.assign(
+						{},
+						grid[canonicalRow + currentRowSpan - 1][colIndex],
+						{ isOriginal: false }
+					);
+				}
+			}
+		}
+	}
+
+	// map new values to each cell in the final grid
+	const finalGrid: IFinalGridCell[][] = [];
+	for (let rowIndex = 0; rowIndex < grid.length; rowIndex++) {
+		finalGrid[rowIndex] = [];
+		for (let colIndex = 0; colIndex < grid[rowIndex].length; colIndex++) {
+			finalGrid[rowIndex][colIndex] = mapFn(
+				grid[rowIndex][colIndex],
+				{ row: rowIndex, col: colIndex },
+				finalGrid
+			);
+		}
+	}
+
+	return finalGrid;
+}
+
+/**
+ * renderRowsWithIdentifiedEdges
+ *
+ * Returns an equivalent list of Tr's where each cell on the perimeter has props set for: `isFirstRow`, `isLastRow`, `isFirstCol`, `isLastCol`, and `isFirstSingle`
+ */
+
+//{renderRowsWithIdentifiedEdges(filterTypes(children, Tr), Th)}
```

Comment:
might be able to remove this comment too

Reply:
good catch. done.